src/js/hooks/useTableDataV2.js

```diff
@@ -45,6 +45,7 @@ const useTableDataV2 = ({
   }));
 
   const fetchData = () => {
+    sourceRef.current = CancelToken.source();
```

Comment:
I think this looks good because this is what we also did for the old `useTableData`. The only request I have is to copy the comment from the `useTableData`, so that everyone understands why it is needed here.